app/javascript/packs/form-steps-wait.tsx

```diff
@@ -1,63 +1,92 @@
 import { render, unmountComponentAtNode } from 'react-dom';
 import { Alert } from '@18f/identity-components';
 
-/**
- * @typedef FormStepsWaitElements
- *
- * @prop {HTMLFormElement} form
- */
+interface FormStepsWaitElements {
+  form: HTMLFormElement;
+}
 
-/**
- * @typedef FormStepsWaitOptions
- *
- * @prop {number} pollIntervalMs Poll interval.
- * @prop {string} waitStepPath URL path to wait step, used in polling.
- * @prop {string=} errorMessage Message to show on unhandled server error.
- * @prop {string=} alertTarget DOM selector of HTML element to which alert should render.
- */
+interface FormStepsWaitOptions {
+  /**
+   * Poll interval.
+   */
+  pollIntervalMs: number;
+
+  /**
+   * URL path to wait step, used in polling.
+   */
+  waitStepPath: string;
+
+  /**
+   * Message to show on unhandled server error.
+   */
+  errorMessage?: string;
 
-/** @type {FormStepsWaitOptions} */
-const DEFAULT_OPTIONS = {
+  /**
+   * DOM selector of HTML element to which alert should render.
+   */
+  alertTarget?: string;
+}
+
+const DEFAULT_OPTIONS: FormStepsWaitOptions = {
   pollIntervalMs: 3000,
   waitStepPath: `${window.location.pathname}_wait`,
 };
 
 /**
  * Returns a DOM document object for given markup string.
  *
- * @param {string} html HTML markup.
+ * @param html HTML markup.
  *
- * @return {Document} DOM document.
+ * @return DOM document.
  */
-export function getDOMFromHTML(html) {
+export function getDOMFromHTML(html: string): Document {
   const dom = document.implementation.createHTMLDocument('');
   dom.body.innerHTML = html;
   return dom;
 }
 
 /**
- * @param {Document} dom
- *
- * @return {boolean} Whether page polls.
+ * @return Whether page polls.
  */
-export function isPollingPage(dom) {
+export function isPollingPage(dom: Document): boolean {
   return Boolean(dom.querySelector('meta[http-equiv="refresh"]'));
```

Comment:
Yeah, I'd actually meant to leave a note here about this as well, since I think now after removing no-JS support for identity verification, it would be a good time to revisit these "wait" steps and have the back-end endpoint return a JSON response structured in the same way I've implemented here for this experience.